plugins/inputs/elasticsearch_query/elasticsearch_query_test.go

```diff
@@ -0,0 +1,682 @@
+package elasticsearch_query
+
+import (
+	"bufio"
+	"context"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/require"
+	elastic5 "gopkg.in/olivere/elastic.v5"
+)
+
+var (
+	testindex = "test-elasticsearch_query-" + strconv.Itoa(int(time.Now().Unix()))
+	setupOnce sync.Once
+	e         = &ElasticsearchQuery{
+		URLs:    []string{"http://" + testutil.GetLocalHost() + ":9200"},
+		Timeout: internal.Duration{Duration: time.Second * 30},
+	}
+)
+
+type esAggregationQueryTest struct {
+	queryName                 string
+	testAggregationQueryInput esAggregation
+	testAggregationQueryData  []aggregationQueryData
+	testMapFields             map[string]string
+	expectedMetrics           []expectedMetric
+	wantBuildQueryErr         bool
+	wantGetMetricFieldsErr    bool
+	wantQueryResErr           bool
+}
+
+type expectedMetric struct {
+	measurement string
+	fields      map[string]interface{}
+	tags        map[string]string
+}
```

Comment:
You might want to use `testutil.MustMetric(measurement, fields, tags, time.Unix(0,0)` instead. This way you can later directly test the metrics using `testutil.RequireMetricsEqual(t, expected, acc.GetTelegrafMetrics(), testutil.IgnoreTime())` where expected is `expected := []telegraf.Metric{ testutil.MustMetric(...), testutil.MustMetric(...),...}`

Reply:
Can you please consider this comment.

Reply:
Done :)